applications/main/infrared/infrared_cli.c

```diff
@@ -177,8 +178,14 @@ static bool infrared_cli_parse_raw(const char* str, InfraredSignal* signal) {
     }
 
     uint32_t* timings = malloc(sizeof(uint32_t) * MAX_TIMINGS_AMOUNT);
-    uint32_t frequency = atoi(frequency_str);
-    float duty_cycle = (float)atoi(duty_cycle_str) / 100;
+    uint32_t frequency;
+    uint32_t duty_cycle_u32;
+    if(strint_to_uint32(frequency_str, NULL, &frequency, 10) != StrintParseNoError ||
+       strint_to_uint32(duty_cycle_str, NULL, &duty_cycle_u32, 10) != StrintParseNoError) {
+        free(timings);
```

Comment:
IMHO just move `malloc` under those checks, it'll save you having to free that allocation, and technically any code size reduction like that is nice.

Reply:
yeah performance is also a concern, I'll fix it

Reply:
Realistically you'll "never" reach this error condition, as in it's probably rare enough that for all intents and purposes it's dead code as far as a regular user is concerned - so it probably doesn't make any difference for performance either way